packages/core/src/messages/responses/cache-list-pop-front.ts

```diff
@@ -1,62 +1,45 @@
 import {SdkError} from '../../errors';
 import {
+  BaseResponseError,
+  BaseResponseMiss,
   ResponseBase,
-  ResponseError,
-  ResponseHit,
-  ResponseMiss,
 } from './response-base';
 import {truncateString} from '../../internal/utils';
+import {CacheListPopFrontResponse} from './enums';
 
 const TEXT_DECODER = new TextDecoder();
 
-/**
- * Parent response type for a list pop front request.  The
- * response object is resolved to a type-safe object of one of
- * the following subtypes:
- *
- * - {Hit}
- * - {Miss}
- * - {Error}
- *
- * `instanceof` type guards can be used to operate on the appropriate subtype.
- * @example
- * For example:
- * ```
- * if (response instanceof CacheListPopFront.Error) {
- *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
- *   // `CacheListPopFront.Error` in this block, so you will have access to the properties
- *   // of the Error class; e.g. `response.errorCode()`.
- * }
- * ```
- */
-export abstract class Response extends ResponseBase {
-  public value(): string | undefined {
-    if (this instanceof Hit) {
-      return (this as Hit).value();
-    }
-    return undefined;
-  }
+interface IResponse {
+  value(): string | undefined;
+  type: CacheListPopFrontResponse;
```

Comment:
I just learned that you can make this value on the interface `readonly`.

```
interface IResponse {
  value(): string | undefined;
  readonly type: CacheListPopFrontResponse;
}
```

 I think we should do that, so we don't accidentally make it mutable inside of the class. I can work on changing the existing ones to be readonly.

Reply:
oh nice! yeah I can update the ones here to be readonly